datasets/io-biodiversity/io_biodiversity.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from datetime import datetime
+from functools import lru_cache
+from pathlib import Path
+from typing import Any, Dict, List, Union
+
+import pystac
+import rasterio
+from pystac.utils import datetime_to_str, str_to_datetime
+from shapely.geometry import shape
+from stactools.core.utils.raster_footprint import RasterFootprint
+
+from pctasks.core.models.task import WaitTaskResult
+from pctasks.core.storage import StorageFactory
+from pctasks.dataset.collection import Collection
+
+NDJSON_PATH = "blob://pcdata01euw/impact/bii-v1/io-vizz-bii.ndjson"
+ASSET_KEY = "data"
+
+handler = logging.StreamHandler()
+handler.setFormatter(logging.Formatter("[%(levelname)s]:%(asctime)s: %(message)s"))
+handler.setLevel(logging.INFO)
+logger = logging.getLogger(__name__)
+logger.addHandler(handler)
+logger.setLevel(logging.INFO)
+
+
+@lru_cache(maxsize=1)
+def get_ndjson(storage_factory: StorageFactory) -> Dict[str, Dict[str, Any]]:
+    storage, path = storage_factory.get_storage_for_file(NDJSON_PATH)
+    item_dicts = storage.read_ndjson(path)
+    return {item["id"]: item for item in item_dicts}
+
+
+class IOBiodiversityIntactness(Collection):
+    @classmethod
+    def create_item(
+        cls, asset_uri: str, storage_factory: StorageFactory
+    ) -> Union[List[pystac.Item], WaitTaskResult]:
+        asset_storage, asset_path = storage_factory.get_storage_for_file(asset_uri)
+
+        asset_href = asset_storage.get_authenticated_url(asset_path)
+        with rasterio.open(asset_href) as src:
+            geometry = RasterFootprint(
+                data_array=src.read_masks(1),
+                crs=src.crs,
+                transform=src.transform,
+                densification_distance=src.transform[0],
+                simplify_tolerance=0.0005,
+                no_data=0,
+            ).footprint()
+
+        if geometry is None:
+            return []
+
+        id = Path(asset_path).stem
+        io_item_dicts = get_ndjson(storage_factory)
+        item_dict = io_item_dicts[id]
```

Comment:
```suggestion
        item_dict = copy.deepcopy(io_item_dicts[id])
```

I have some _slight_ discomfort with mutating the result here, since it seems to be coming out of an `lru_cache`. Granted, IIUC we're only mutation something under the `id` key which *should* only happen once per processes, but probably safer to copy.

Reply:
That's a good point. I'll make that change.